[ADD] sales: adding pdf quote builder doc for 16/17 (#6029)
odoo/documentation: content/applications/sales/sales/send_quotations/pdf_quote_builder.rst

=================
PDF quote builder
=================

The *PDF Quote Builder* in Odoo *Sales* provides the opportunity to export a fully-custom PDF file
for quotes, showcasing various information and design elements, instead of just the price and total.

The PDF Quote Builder can be used to add header pages, product descriptions, and footer pages to a
quote.

Having a customized PDF in quotes provides a heightened conclusion to the shopping experience for
customers, and adds an elegant level of professionalism to a company.

Configuration
=============

In order to add custom PDF files for quotes, the :guilabel:`PDF Quote builder` feature *must* be
configured.

To do that, navigate to :menuselection:`Sales app --> Configuration --> Settings`. Then, on the
:guilabel:`Settings` page, scroll to the :guilabel:`Quotations & Orders` section, and locate the
:guilabel:`PDF Quote builder` feature.

.. image:: pdf_quote_builder/pdf-quote-builder-feature.png
   :align: center

>    :alt: The PDF Quote Builder feature located on the Settings page of the Sales application.

Author: Great point! I'll change it to this image right away. Thanks for providing such an awesome screenshot to accompany your comment. Super helpful! 👍

builder` section of the *Sales* :guilabel:`Settings` page, be sure to click :guilabel:`Save`.

The files uploaded here will be the default PDF used for all quotes.

.. note::
    Values set here are company-specific.

Dynamic text in PDFs
====================

While creating custom PDFs for quotes, using any standard PDF editor software, certain dynamic texts
should be utilized throughout the design of the PDF to ensure Odoo can automatically fill in the
appropriate content accordingly.

Dynamic text values are, essentially, coded placeholders that Odoo fills in automatically, according
to the specific information related to the quote.

Dyanmic text values
-------------------

Below are the dynamic text values and what they will represent in the custom PDFs:

- :guilabel:`name`: Order Reference
- :guilabel:`partner_id_name`: Customer Name
- :guilabel:`user_id_name`: Salesperson Name
- :guilabel:`amount_untaxed`: Untaxed Amount
- :guilabel:`amount_total`: Total Amount
- :guilabel:`delivery_date`: Delivery Date
- :guilabel:`validity_date`: Expiration Date
- :guilabel:`client_order_ref`: Customer Reference

> 

Author: I completely understand your logic here. However, I think this may be a bit "disconnected" from the overall concept of this doc, aimed specifically at PDFs (not emails, per se). That said, again, I fully get where you're coming from with this. So, I think I'll add a quick visual example of a PDF using dynamic text placeholders to showcase a more doc-appropriate sample of how *that* would look. Hope that will suffice. Thanks for bringing this to my attention. I really appreciate your thoughtfulness (and thoroughness) with your suggestions. 👍

> 

Reviewer: Aww (: Thanks for taking my feedback seriously.The screenshot is perfect, and I now have a clearer understanding of how dynamic text is used in a PDF. Thank you for addressing my questions!

Once the PDF file(s) are complete, save them to the computer's hard drive, and proceed to upload

> them to Odoo, via :menuselection:`Sales app --> Configuration --> Settings --> PDF Quote builder -->

Author: Well, actually, you *did* find it -- as this is the exact place I was intended the reader to find/upload their PDFs...so I'll make sure to clarify that in the doc. Thanks for the suggestion.

Save`.

Add PDF to product
==================

In Odoo *Sales*, it's also possible to add a custom PDF to a product. When a PDF is added to a
product, and that product is used in a quotation, that PDF is also inserted in the final PDF.

To add a custom PDF to a product, start by navigating to :menuselection:`Sales app --> Products -->
Products`, and selected the desired product to which a custom PDF should be added.

> 

Reviewer 2: A document could also be added to a product variant instead of a product. Be aware that if there are documents on a product AND on its variant, only the documents in the variant will be shown. =]

Author: Thanks for the reminder. I added this as a note admonition right below this. 👍

to the specific information related to the quote.

Below are the dynamic text values and what they will represent in the custom PDFs:

- :guilabel:`name`: Order Reference
- :guilabel:`partner_id_name`: Customer Name
- :guilabel:`user_id_name`: Salesperson Name
- :guilabel:`amount_untaxed`: Untaxed Amount
- :guilabel:`amount_total`: Total Amount
- :guilabel:`delivery_date`: Delivery Date
- :guilabel:`validity_date`: Expiration Date
- :guilabel:`client_order_ref`: Customer Reference

Product-specific dynamic text values are as follows:

- :guilabel:`description`: Product Description
- :guilabel:`quantity`: Quantity
- :guilabel:`uom`: Unit of Measure (UoM)
- :guilabel:`price_unit`: Price Unit
- :guilabel:`discount`: Discount
- :guilabel:`product_sale_price`: Product List Price
- :guilabel:`taxes`: Taxes name joined by a comma (`,`)
- :guilabel:`tax_excl_price`: Tax Excluded Price
- :guilabel:`tax_incl_price`: Tax Included Price

PDF quote
=========

When a quotation with a configured PDF-related quote has been confirmed, click the :guilabel:`⚙️
(gear)` icon to reveal a drop-down menu. From this drop-down menu, select :guilabel:`Print`, then

> select :guilabel:`PDF Quote`.

Reviewer: I'm slightly confused by this sentence. This is how I think I'm following what you're saying, and let me know if my interpretation is incorrect:

After using the custom PDF builder to personalize a quotation, *and then* we sent the quote to the customer, *and then* the customer placed the order through the Customer Portal...
you can print the PDF quote with these following steps?

I'm not following why this is really useful? Could you elaborate a little more? (:

Author: You bring up a great point. And, frankly, I agree -- I'm not certain how this is useful. To be clear, I only added this at the bottom because it was mentioned by JCS in the initial Project Task requesting this doc to be made, so I felt obligated to throw it in towards the end.

So, given that JCS made this mention directly, I think I'll opt to keep it in -- however, I'll be sure to clean up the language, and add more of a "reasoning" behind a user (potentially) needing/wanting to do this.

Thanks for pointing this out!

.. image:: pdf_quote_builder/drop-down-print-pdf.png
   :align: center
   :alt: Print pdf quote option on drop-down menu located on confirmed sales order in Odoo Sales.

Doing so instantly downloads the PDF quote. When opened, the PDF quote, along with the configured

> product PDF that was set to be visible inside the quote, can be seen.

Reviewer: I'm still a little unsure why it's useful to see the quote. Is it to confirm that the dynamic placeholders work as intended?

   :guilabel:`partner_id_name`). When uploaded into the database, Odoo auto-populates those fields
   with the information from their respective fields.

   In this case, Odoo would auto-populate the Sales Order Reference in the :guilabel:`name` dynamic
   text field, and the Customer Name in the :guilabel:`partner_id_name` field.

   .. image:: pdf_quote_builder/pdf-quote-builder-sample.png
      :align: center
      :alt: PDF quote being built using common dynamic placeholders.

Once the PDF file(s) are complete, save them to the computer's hard drive, and proceed to upload
them to Odoo via :menuselection:`Sales app --> Configuration --> Settings --> PDF Quote builder`.

Upload the created PDF in the :guilabel:`Header pages` or :guilabel:`Footer pages` field.

Once the upload(s) are complete, click :guilabel:`Save`.

Add PDF to product
==================

In Odoo *Sales*, it's also possible to add a custom PDF to a product form. When a PDF is added to a
product, and that product is used in a quotation, that PDF is also inserted in the final PDF.

To add a custom PDF to a product, start by navigating to :menuselection:`Sales app --> Products -->

> Products`, and selected the desired product to which a custom PDF should be added.

Reviewer 3: ```suggestion
Products`, and select the desired product to which a custom PDF should be added.
```

.. note::
   A document could also be added to a product variant, instead of a product. If there are documents

>    on a product *and* on its variant, **only** the documents in the variant are shown.

Reviewer 3: Can this doc include brief instructions on how to attach a document to a product *variant* as well?

On the product page, click the :guilabel:`Documents` smart button at the top of the page.

.. image:: pdf_quote_builder/documents-smart-button.png
   :align: center
   :alt: The Documents smart button on a product form in Odoo Sales.

Doing so reveals a separate :guilabel:`Documents` page for that product, wherein files related to
that product can be uploaded. From this page, either click :guilabel:`New` or :guilabel:`Upload`.

Clicking :guilabel:`Upload` instantly provides the opportunity to upload the desired document. Then,
the document can be further configured on the document card, or by clicking the three dots icon in
the top right corner of the document card, and then clicking :guilabel:`Edit`.

> Clicking :guilabel:`New` reveals a blank documents form, in which the desired PDF can be uploaded,
> as well, via the :guilabel:`Upload your file` button on the form, located in the :guilabel:`File

Reviewer 3: ```suggestion
Clicking :guilabel:`New` reveals a blank documents form, in which the desired PDF can be uploaded
via the :guilabel:`Upload your file` button on the form, located in the :guilabel:`File
```

Content` field.

.. image:: pdf_quote_builder/blank-document-form.png
   :align: center
   :alt: A standard document form with various fields for a specific product in Odoo Sales.

Various information and configurations related to the uploaded document can be modified here.

> The first field on the documents form is for the :guilabel:`Name` of the document, and it's
> greyed-out until a document is uploaded. Once a PDF has been uploaded, the :guilabel:`Name` field is

Reviewer 3: ```suggestion
The first field on the documents form is for the :guilabel:`Name` of the document, and it is
greyed-out (not clickable) until a document is uploaded. Once a PDF has been uploaded, the :guilabel:`Name` field is
```

.. example::
   When the :guilabel:`Show on product page` option is enabled, a link to the uploaded document,
   `Sample Builder.pdf`, appears on the product's page, located on the frontend in the online store.

   It appears beneath a :guilabel:`Documents` heading, with a link showcasing the name of the
   uploaded document.

    .. image:: pdf_quote_builder/show-product-page.png
       :align: center
       :alt: Showing a link to an uploaded document on a product page using Odoo Sales.

PDF quote
=========

Once a quote with a pre-configured PDF has been confirmed, Odoo provides the option to print the
confirmed quote to check for errors, or to keep for records.

To print the PDF quote, navigate to the confirmed quote, and click the :guilabel:`⚙️ (gear)` icon to
reveal a drop-down menu. From this drop-down menu, select :guilabel:`Print`, then select
:guilabel:`PDF Quote`.

.. image:: pdf_quote_builder/drop-down-print-pdf.png
   :align: center
   :alt: Print pdf quote option on drop-down menu located on confirmed sales order in Odoo Sales.

Doing so instantly downloads the PDF quote. When opened, the PDF quote, along with the configured

> product PDF that was set to be visible inside the quote, can be viewed and printed.

Reviewer 3: I agree it would be great to include an example of a finished quote with an attached PDF here at the end.

   different variations per template.

Clicking the :guilabel:`🗑️ (trash)` icon deletes the current PDF file and replaces the blank field
with an :guilabel:`Upload your file` button.

Once the desired PDF file(s) are uploaded in the appropriate fields in the :guilabel:`PDF Quote
builder` section of the *Sales* :guilabel:`Settings` page, be sure to click :guilabel:`Save`.

The files uploaded here will be the default PDF used for all quotes.

.. note::
   Values set in the PDF Quote Builder settings are company-specific.

Dynamic text in PDFs
====================

While creating custom PDFs for quotes, use *dynamic text* for Odoo to auto-fill the PDF content with
information related to the quote from the Odoo database, like names, prices, etc.

Dynamic text values are form components (text inputs) that can be added in a PDF file, and Odoo
automatically fills those values in with information related to the quote.

Dynamic text values
-------------------

Below are common dynamic text values used in custom PDFs, and what they represent:

- :guilabel:`name`: Sales Order Reference
- :guilabel:`partner_id_name`: Customer Name

> - :guilabel:`user_id_name`: Salesperson Name

Reviewer 4: there should be 2 underscore between id and name. partner_id__name. 
Why a the double underscore ? because it replaces the '.' in partner_id.name and user_id.name (name of these field in ORM)
Unfortunately the library does not support ".". 
We can mention somewhere in the doc this double underscore to replace the "."

Author: I'll make these changes right away in another PR -- considering this one was already closed. I'll link the new PR in a comment here when it's completed. Thanks for pointing this out. I really appreciate it! 👍

Author: @dalaOdoo https://github.com/odoo/documentation/pull/6322

to create a detailed quote. It can also inject dynamic texts in the PDF to personalize the offer
for the customer.

Having a customized PDF in quotes provides a heightened conclusion to the shopping experience for
customers, and adds an elegant level of professionalism to a company.

Configuration
=============

In order to add custom PDF files for quotes, the :guilabel:`PDF Quote builder` feature *must* be
configured.

To do that, navigate to :menuselection:`Sales app --> Configuration --> Settings`. Then, on the
:guilabel:`Settings` page, scroll to the :guilabel:`Quotations & Orders` section, and locate the
:guilabel:`PDF Quote builder` feature.

.. image:: pdf_quote_builder/pdf-quote-builder-feature.png
   :align: center
   :alt: The PDF Quote Builder feature located on the Settings page of the Sales application.

Here, custom :guilabel:`Header pages` and :guilabel:`Footer pages` can be uploaded. To upload
either, click the :guilabel:`Upload your file` button, or the :guilabel:`✏️ (pencil)` icon to the
right of the desired field, and proceed to locate, select, and upload the desired PDF file.

> .. note::
>    Headers and footers can also be added directly in a quotation template, so it's possible to have
>    different headers/footers on a quotation template(s).

Reviewer 5: Clarity
```suggestion
.. note::
   Headers and footers can also be added directly in a quotation template, so it's possible to have
   different variations per template.
```

Clicking the :guilabel:`🗑️ (trash)` icon deletes the current PDF file and replaces the blank field
with an :guilabel:`Upload your file` button.

Once the desired PDF file(s) are uploaded in the appropriate fields in the :guilabel:`PDF Quote
builder` section of the *Sales* :guilabel:`Settings` page, be sure to click :guilabel:`Save`.

The files uploaded here will be the default PDF used for all quotes.

.. note::
   Values set in the PDF Quote Builder settings are company-specific.

Dynamic text in PDFs
====================

While creating custom PDFs for quotes, use *dynamic text* for Odoo to auto-fill the PDF content with
information related to the quote from the Odoo database, like names, prices, etc.

> Dynamic text values are form components (text inputs) that can be added in a PDF file, and Odoo
> automatically fills it in with information related to the quote. Odoo fills these components
> according to their name. Doing so streamlines the writing of personalized quotations.

Reviewer 5: - Mixed subject in the first sentence
- don't need 2nd or 3rd sentence after that

```suggestion
Dynamic text values are form components (text inputs) that can be added in a PDF file, and Odoo
automatically fills those values in with information related to the quote.
```

Dynamic text values
-------------------

Below are common dynamic text values used in custom PDFs, and what they represent:

> - :guilabel:`name`: Sales Order Reference

Reviewer 5: think these lists would be better formatted as a table using `pre` formatting instead of guis but we can do that later if we want 😉

- :guilabel:`partner_id_name`: Customer Name
- :guilabel:`user_id_name`: Salesperson Name
- :guilabel:`amount_untaxed`: Untaxed Amount
- :guilabel:`amount_total`: Total Amount
- :guilabel:`delivery_date`: Delivery Date
- :guilabel:`validity_date`: Expiration Date
- :guilabel:`client_order_ref`: Customer Reference

Product-specific dynamic text values are as follows:

- :guilabel:`description`: Product Description
- :guilabel:`quantity`: Quantity
- :guilabel:`uom`: Unit of Measure (UoM)
- :guilabel:`price_unit`: Price Unit
- :guilabel:`discount`: Discount
- :guilabel:`product_sale_price`: Product List Price
- :guilabel:`taxes`: Taxes name joined by a comma (`,`)
- :guilabel:`tax_excl_price`: Tax Excluded Price
- :guilabel:`tax_incl_price`: Tax Included Price

.. example::

>    When a PDF is built, it's best to use common dynamic text values (:guilabel:`name` and

Reviewer 5: ```suggestion
   When a PDF is built, it's best practice to use common dynamic text values (:guilabel:`name` and
```

.. image:: pdf_quote_builder/documents-smart-button.png
   :align: center
   :alt: The Documents smart button on a product form in Odoo Sales.

Doing so reveals a separate :guilabel:`Documents` page for that product, wherein files related to
that product can be uploaded. From this page, either click :guilabel:`New` or :guilabel:`Upload`.

Clicking :guilabel:`Upload` instantly provides the opportunity to upload the desired document. Then,
the document can be further configured on the document card, or by clicking the three dots icon in
the top right corner of the document card, and then clicking :guilabel:`Edit`.

Clicking :guilabel:`New` reveals a blank documents form, in which the desired PDF can be uploaded
via the :guilabel:`Upload your file` button on the form, located in the :guilabel:`File Content`
field.

.. image:: pdf_quote_builder/blank-document-form.png
   :align: center
   :alt: A standard document form with various fields for a specific product in Odoo Sales.

Various information and configurations related to the uploaded document can be modified here.

The first field on the documents form is for the :guilabel:`Name` of the document, and it is

> greyed-out (not clickable) until a document is uploaded. Once a PDF has been uploaded, the

Reviewer 5: nitpick
```suggestion
grayed-out (not clickable) until a document is uploaded. Once a PDF has been uploaded, the
```

for quotes, showcasing various information and design elements, instead of just the price and total.

The PDF Quote Builder can be used to add header pages, product descriptions, and footer pages to a
quote.

Having a customized PDF in quotes provides a heightened conclusion to the shopping experience for
customers, and adds an elegant level of professionalism to a company.

Configuration
=============

In order to add custom PDF files for quotes, the :guilabel:`PDF Quote builder` feature *must* be
configured.

To do that, navigate to :menuselection:`Sales app --> Configuration --> Settings`. Then, on the
:guilabel:`Settings` page, scroll to the :guilabel:`Quotations & Orders` section, and locate the
:guilabel:`PDF Quote builder` feature.

.. image:: pdf_quote_builder/pdf-quote-builder-feature.png
   :align: center
   :alt: The PDF Quote Builder feature located on the Settings page of the Sales application.

Here, custom :guilabel:`Header pages` and :guilabel:`Footer pages` can be uploaded. To upload
either, click the :guilabel:`Upload your file` button, or the :guilabel:`✏️ (pencil)` icon to the

> right of the desired field, and proceed to locate, select, and upload the desired PDF file.

Reviewer 2: Header en footer could also directly be added in a quotation template, so you could have different ones on a/some template(s) ^^

Author: Got it. Just added a note admonition to include this. Thanks for the suggestion. 👍

=================
PDF quote builder
=================

The *PDF Quote Builder* in Odoo *Sales* provides the opportunity to export a fully-custom PDF file
for quotes, showcasing various information and design elements, instead of just the price and total.

The PDF Quote Builder can be used to add header pages, product descriptions, and footer pages to a
quote.

Having a customized PDF in quotes provides a heightened conclusion to the shopping experience for

> customers, and adds an elegant level of professionalism to a company.

Author: I think this quick sentence adds a bit of context to any curious reader who may have just wandered into this doc, wondering what the benefits of a custom PDF in a quote could be...but, if you think it would be best to remove it, just let me know @meng-odoo and I'll get rid of it right away.

Configuration
=============

In order to add custom PDF files for quotes, the :guilabel:`PDF Quote builder` feature *must* be
configured.

To do that, navigate to :menuselection:`Sales app --> Configuration --> Settings`. Then, on the
:guilabel:`Settings` page, scroll to the :guilabel:`Quotations & Orders` section, and locate the
:guilabel:`PDF Quote builder` feature.

.. image:: pdf_quote_builder/pdf-quote-builder-feature.png
   :align: center
   :alt: The PDF Quote Builder feature located on the Settings page of the Sales application.

Here, custom :guilabel:`Header pages` and :guilabel:`Footer pages` can be uploaded. To upload
either, click the :guilabel:`Upload your file` button, or the :guilabel:`✏️ (pencil)` icon to the
right of the desired field, and proceed to locate, select, and upload the desired PDF file.

.. note::
   Header and footers can also be added directly in a quotation template, so it's possible to have

>    different headers/footers on a template(s).

Author: It's my understanding that headers and footers can be added to a quotation template to have *different* header/footer than the original/default--not necessarily *multiple* ones...if that makes sense haha

Clicking the :guilabel:`🗑️ (trash)` icon deletes the current PDF file, and replaces the blank field
with an :guilabel:`Upload your file` button. When :guilabel:`Upload your file` is clicked, proceed
to upload the desired PDF files, using the same process detailed above.

Once the desired PDF file(s) are uploaded in the appropriate fields in the :guilabel:`PDF Quote
builder` section of the *Sales* :guilabel:`Settings` page, be sure to click :guilabel:`Save`.

The files uploaded here will be the default PDF used for all quotes.

.. note::
   Values set here are company-specific.

Dynamic text in PDFs
====================

> While creating custom PDFs for quotes, use dynamic text for Odoo to auto-fill PDF content, like

Reviewer 4: ```suggestion
While creating custom PDFs for quotes, use *PDF form* with components (Dynamic texts) for Odoo to auto-fill PDF content with information related to the quote, like
```

names and prices.

> Dynamic text values are, essentially, coded placeholders that Odoo fills in automatically to

Reviewer 4: ```suggestion
Dynamic text values are, essentially, form components (text inputs) you can add in your PDF file and Odoo will automatically fill it with information related to the quote. Odoo will fill these components accordingly to their name. 
```

streamline the writing of personalized quotations.

> Dynamic text values
> -------------------
> 
> Below are common dynamic text values used in custom PDFs, and what they represent:

Reviewer 4: ```suggestion
Authorized form components names in Header and Footer pages
-------------------

Below is the list of form components names you can use in Header and Footer pages and what they will represent in the custom PDFs:
```

- :guilabel:`name`: Sales Order Reference
- :guilabel:`partner_id_name`: Customer Name
- :guilabel:`user_id_name`: Salesperson Name
- :guilabel:`amount_untaxed`: Untaxed Amount
- :guilabel:`amount_total`: Total Amount
- :guilabel:`delivery_date`: Delivery Date
- :guilabel:`validity_date`: Expiration Date
- :guilabel:`client_order_ref`: Customer Reference

.. example::
   A sample PDF being built using common dynamic text values (:guilabel:`name` and

>    :guilabel:`partner_id_name`).

Author: Great point. I did my best to elaborate on this as succinctly as possible to describe things a bit further for those who cannot view the image.

that product can be uploaded. From this page, either click :guilabel:`New` or :guilabel:`Upload`.

Clicking :guilabel:`Upload` instantly provides the opportunity to upload the desired document.

.. note::
    If :guilabel:`Upload` is clicked from the :guilabel:`Documents` page, Odoo uploads the desired
    file, and reveals a documents form, with that file already loaded in the :guilabel:`File
    Content` field.

    In other words, clicking :guilabel:`Upload` directly from the :guilabel:`Documents` page results
    in a documents form that can be further configured and customized.

Clicking :guilabel:`New` reveals a blank documents form, in which the desired PDF can be uploaded,
as well, via the :guilabel:`Upload your file` button on the form, located in the :guilabel:`File
Content` field.

.. image:: pdf_quote_builder/blank-document-form.png
   :align: center
   :alt: A standard document form with various fields for a specific product in Odoo Sales.

Various information and configurations related to the uploaded document can be modified here.

The first field on the documents form is for the :guilabel:`Name` of the document, and it's

> greyed-out. If a PDF has already been uploaded, or will be uploaded via the :guilabel:`Upload your
> file` button on the form, the :guilabel:`Name` field is auto-populated with the name of the PDF.

Reviewer 3: ```suggestion
greyed-out until a document is uploaded. Once a PDF has been uploaded, the :guilabel:`Name` field is auto-populated with the name of the PDF, and it can then be edited.
```

> Next, prior to uploading a document, there's the option to designate whether the document is a

Reviewer 3: ```suggestion
Prior to uploading a document, there's the option to designate whether the document is a
```

:guilabel:`File` or :guilabel:`URL` from the :guilabel:`Type` drop-down field menu.

.. image:: pdf_quote_builder/document-form-uploaded-pdf.png
   :align: center
   :alt: A standard document form with an uploaded pdf in Odoo Sales.

.. note::
    If a PDF is uploaded, the :guilabel:`Type` field is auto-populated to :guilabel:`File`, and it
    cannot be modified.

Then, in the :guilabel:`Sales` section, in the :guilabel:`Visible at` field, click the drop-down
menu, and select either: :guilabel:`Quotation`, :guilabel:`Confirmed order`, or :guilabel:`Inside
quote`.

- :guilabel:`Quotation`: the document is sent to (and accessible by) customers at any time.

- :guilabel:`Confirmed order`: the document is sent to customers upon the confirmation of an order.
  This is best for user manuals and other supplemental documents.

- :guilabel:`Inside quote`: the document is included in the PDF of the quotation, between the header

>   pages and the quote table.

Reviewer 3: ```suggestion
  pages and the :guilabel:`Pricing` section of the quote.
```

.. example::
   When the :guilabel:`Inside quote` option for the :guilabel:`Visible at` field is chosen, and the
   custom PDF file, `Sample Builder.pdf` is uploaded, the PDF is visible on the quotation the in the
   *customer portal* under the :guilabel:`Documents` field.

    .. image:: pdf_quote_builder/pdf-inside-quote-sample.png
       :align: center
       :alt: Sample of an uploaded pdf with the inside quote option chosen in Odoo Sales.

Lastly, in the :guilabel:`E-Commerce` section, decide whether or not to :guilabel:`Show on product
page` on the front-end (in the online store).

.. example::
   When the :guilabel:`Show on product page` option is enabled, a link to the uploaded document,
   `Sample Builder.pdf`, appears on the product's page, located on the frontend in the online store.

   It appears beneath a :guilabel:`Documents` heading, with a link showcasing the name of the
   uploaded document.

    .. image:: pdf_quote_builder/show-product-page.png
       :align: center
       :alt: Showing a link to an uploaded document on a product page using Odoo Sales.

> Dynamic text values

Reviewer 4: ```suggestion
Authorized form components names in Product descriptions
```

-------------------

While creating custom PDFs for quotes, using any standard PDF editor software, certain dynamic texts
should be utilized throughout the design of the PDF to ensure Odoo can automatically fill in the
appropriate content accordingly.

Dynamic text values are, essentially, coded placeholders that Odoo fills in automatically, according
to the specific information related to the quote.

> Below are the dynamic text values and what they will represent in the custom PDFs:

Reviewer 4: ```suggestion
Below is the list of authorized names you can use in product descriptions and what they will represent in the custom PDFs:
```

- :guilabel:`name`: Sales Order Reference
- :guilabel:`partner_id_name`: Customer Name
- :guilabel:`user_id_name`: Salesperson Name
- :guilabel:`amount_untaxed`: Untaxed Amount
- :guilabel:`amount_total`: Total Amount
- :guilabel:`delivery_date`: Delivery Date
- :guilabel:`validity_date`: Expiration Date

> - :guilabel:`client_order_ref`: Customer Reference

Reviewer 3: I would delete this whole section from this part of the doc, and just include the "Product-specific dynamic text values" list in the above "Dynamic text in PDFs" H2 section.

But if you decide to keep this section here, you probably want change the intro paragraph to match the one in the "Dynamic text in PDFs" H2 section, since the intro paragraph here seems to be an older version.

Reviewer 4: I agree, I suggest to change the title of the section and only keep the list of authorized name for form components

Product-specific dynamic text values are as follows:

- :guilabel:`description`: Product Description
- :guilabel:`quantity`: Quantity
- :guilabel:`uom`: Unit of Measure (UoM)
- :guilabel:`price_unit`: Price Unit
- :guilabel:`discount`: Discount
- :guilabel:`product_sale_price`: Product List Price
- :guilabel:`taxes`: Taxes name joined by a comma (`,`)
- :guilabel:`tax_excl_price`: Tax Excluded Price
- :guilabel:`tax_incl_price`: Tax Included Price

PDF quote
=========

> On a quote, with a pre-configured PDF, has been confirmed, Odoo provides the option to print the

Reviewer 3: ```suggestion
Once a quote with a pre-configured PDF has been confirmed, Odoo provides the option to print the
```

confirmed quote to check for errors, or to keep for records.

> To do that, navigate to the desired confirmed quote, and click the :guilabel:`⚙️ (gear)` icon to

Reviewer 3: ```suggestion
To print the PDF quote, navigate to the confirmed quote, and click the :guilabel:`⚙️ (gear)` icon to
```
